n2survey/lime/survey.py

```diff
@@ -280,6 +315,75 @@ def get_responses(
 
         return responses
 
+    def __getitem__(
+        self, key: Union[pd.Series, pd.DataFrame, str, list, tuple]
+    ) -> Union["LimeSurvey", pd.DataFrame]:
+        """Retrieve or slice the responses DataFrame
+
+        Args:
+            key (pd.Series, pd.DataFrame, str, list, or tuple): A key for
+                DataFrame slicing or get_responses method
+
+        Returns:
+            LimeSurvey: A copy of LimeSurvey instance with filtered responses
+                DataFrame
+        """
+        filtered_survey = self.__copy__()
+
+        # A bool-valued Series, e.g. survey[survey.responses["A3"] == "A5"]
+        if isinstance(key, (pd.Series, pd.DataFrame)):
+            filtered_survey.responses = filtered_survey.responses[key]
+        # A question id, e.g. survey["A3"]
+        elif isinstance(key, str):
+            columns = filtered_survey.get_question(key).index
+            filtered_survey.responses = filtered_survey.responses[columns]
+        # A list of columns, e.g. survey[["C3_SQ001", "C3_Sq002"]]
+        elif isinstance(key, list):
+            columns = [
+                column
+                for question in key
+                for column in filtered_survey.get_question(question).index.to_list()
+            ]
+            filtered_survey.responses = filtered_survey.responses[columns]
+        # Two args, e.g. survey[survey.responses["A3"] == "A5", "B1"]
+        # or survey[1:10, ["B1", "C1_SQ001"]]
+        elif isinstance(key, tuple):
+            assert len(key) == 2
```

Comment:
This assertion can be confusing for a user. Rather we can combine the condition with the one in `elif`
```suggestion
        elif isinstance(key, tuple) and len(key)==2:
```